ilastik/applets/dataSelection/addFileButton.py

```diff
@@ -61,6 +61,7 @@ def __init__(self, parent, new=False):
             "/../../shell/gui/icons/16x16/actions/list-add.png"),
             "Add..." if new == False else "Add New...", parent)
 
+        self.index = index
```

Comment:
We have discussed this line a few weeks ago if I recall and I'm opting to quote the zen of python here:

```
readability counts
```

Please don't use properties in the constructor. For me properties are to interact with the object from the outside. It's kind of weird to do it from within.

also, with the default argument `None` this produces an invalid object (accessing `index` afterwards will fail). What are you trying to do achieve with the type check for `QModelIndex`? Setting it to a different type will leave the Object in incomplete state (`__index` is not set).

Reply:
I dont know if the next commit resolves this. Do you suggest to make _index a 'public' attribute and get rid of getter/setter?